### PR TITLE
ctags: Use C99 bool instead of defining our own

### PR DESCRIPTION
**Pull request by collinfunk**
Use stdbool.h definitions instead of defining non-standard ones.

### CODE REVIEW

**Comment by collinfunk** [comment]
Two of the style(9) failures are from the weird GETC macro. The others are because I did not add previously missing parentheses around return values. Should I amend or leave it for a different patch?

**Comment by bsdimp** [comment]
I'd match the existing style, so don't change the returns.
The GETC stuff is a weird macro, and it's just a false positive, so it's good.

**Comment by bsdimp** [comment]
I think this is ready. I'll try to land it later today.

File: usr.bin/ctags/fortran.c [hunk 1]
```diff
@@ -48,7 +48,7 @@ PF_funcs(void)
 	char	*cp;
 	char	tok[MAXTOKEN];
 
-	for (pfcnt = NO;;) {
+	for (pfcnt = false;;) {
```
Comment (markpeek) [inline usr.bin/ctags/fortran.c:51]: Should `int PF_funcs(void)` be changed to `bool PF_funcs(void)` due to these changes?

Comment (bsdimp) [reply usr.bin/ctags/fortran.c:51]: I think so because we're return pfcnt 3 lines below.

Comment (collinfunk) [reply usr.bin/ctags/fortran.c:51]: Thanks for checking, you are right. While fixing it I also noticed that "skip_key" in C.c and ctags.h should return bool. I'll amend those changes to the commit.

**Comment by collinfunk** [comment]
That should be all of them hopefully...

**Comment by markjdb** [comment]
Merged.

**Pull request closed without merging**